src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/EventGridEventExtensions.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Messaging.EventGrid;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+	/// <summary>
+	/// Extension of the EventGridEvent class for creating <see cref="PushNotification"/> object from an <see cref="EventGridEvent "/> object.
+	/// </summary>
+	public static class EventGridEventExtensions
+    {
+		private const string SyncTokenPropertyName = "syncToken";
+
+		/// <summary>
+		/// This method uses an EventGridEvent from EventGrid and tries to create a <see cref="PushNotification"/>
+		/// </summary>
+		/// <param name="eventGridEvent"> EventGridEvent from EventGrid</param>
+		/// <param name="pushNotification"> out parameter set up in this method</param>
+		/// <returns></returns>
+		public static bool TryCreatePushNotification(this EventGridEvent eventGridEvent, out PushNotification pushNotification)
+		{
+			pushNotification = null;
+
+			if (eventGridEvent == null || eventGridEvent.Data == null || eventGridEvent.EventType == null || eventGridEvent.Subject == null)
```

Comment:
The typical expectation for extension methods is not to check for null on the object you are extending.

If we look at this method, the expected usage is
```
eventGridEvent.TryCreatePushNotification(out PushNotification pushNotification);
```

If the method was an instance method instead of an extension method that would throw a null-ref if `eventGridEvent` was null. Since extension methods are used the same way they share the same expectation.